### PR TITLE
Add support for Debian 12

### PR DESCRIPTION
**Pull request by Justin-Fernbaugh**
PR for adding Debian 12 OS support to this cookbook.

### CODE REVIEW

**Comment by ramereth** [comment]
A few issues with this:
- Debian should be excluded from the IUS suite/recipe as this is only for EL platforms
- The gitlfs suite fails due to the fact that the test recipe uses `nobody`. Debian uses `nogroup` for the group and does not have a `nobody` user.

File: test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb [hunk 1]
```diff
@@ -19,13 +19,25 @@
 # This recipe creates a git_credentials resource for testing in ChefSpec.
 # Attributes allow changing values for different test contexts.
 
+
+pseudoGroup = 'nobody'
```
Comment (ramereth) [inline test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb:21]:
```suggestion
nogroup = platform_family?('debian') ? 'nogroup' : 'nogroup'
```

File: test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb [hunk 2]
```diff
@@ -19,13 +19,25 @@
...
+if platform_family?('debian')
+  pseudoGroup = 'nogroup' 
+
+  directory '/tmp/bar' do
+    owner 'nobody'
+    group 'nogroup'
+    mode '0755'
+  end
+end
+
```
Comment (ramereth) [inline test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb]: This whole block can be removed by what I did above. I'm confused why you're creating the directory however.

File: test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb [hunk 3]
```diff
@@ -19,13 +19,25 @@
...
 osl_gitlfs '/foo' do
   repository 'https://git.osuosl.org/osuosl/test-lfs.git'
 end
 
 osl_gitlfs '/tmp/bar' do
   repository 'https://git.osuosl.org/osuosl/test-lfs.git'
   user 'nobody'
-  group 'nobody'
+  group pseudoGroup
```
Comment (ramereth) [inline test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb:39]:
```suggestion
  group nogroup
```

File: test/integration/osl-gitlfs/inspec/osl_gitlfs_spec.rb [hunk 4]
```diff
@@ -1,3 +1,6 @@
+pseudoGroup = 'nobody'
+pseudoGroup = 'nogroup' if os.family == 'debian'
```
Comment (ramereth) [inline test/integration/osl-gitlfs/inspec/osl_gitlfs_spec.rb:2]: Do the same as I suggested above please.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb (hunk 2 above)
Comment (Justin-Fernbaugh) [reply test/cookbooks/osl-git-test/recipes/osl_gitlfs.rb]: The /tmp/bar directory was getting "dubious ownership" git errors when switching from group 'nobody' to 'nogroup' which cancelled the command execution. This is just a way I found around that error

File: test/integration/osl-gitlfs/inspec/osl_gitlfs_spec.rb (hunk 4 above)
Comment (Justin-Fernbaugh) [reply test/integration/osl-gitlfs/inspec/osl_gitlfs_spec.rb]: Pushed changes

File: Berksfile [hunk 5]
```diff
@@ -3,6 +3,7 @@ source 'https://supermarket.chef.io'
 cookbook 'osl-repos', git: 'git@github.com:osuosl-cookbooks/osl-repos'
 cookbook 'osl-resources', git: 'git@github.com:osuosl-cookbooks/osl-resources', branch: 'main'
 cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux'
+cookbook 'apt', git: 'git@github.com:sous-chefs/apt', branch: 'main'
```
Comment (Justin-Fernbaugh) [inline Berksfile:6]: Pull from supermarket instead remove git branch

Comment (Justin-Fernbaugh) [reply Berksfile:6]: Pushed

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you please update `default['osl-git']['ius_git_package']` to be `git226`? It seems `git224` was replaced by 226.

**Comment by Justin-Fernbaugh** [comment]
> Can you please update `default['osl-git']['ius_git_package']` to be `git226`? It seems `git224` was replaced by 226.

Updated the package to be git226 and updated the tests as well

File: attributes/default.rb [hunk 6]
```diff
@@ -2,4 +2,4 @@
 default['osl-git']['secrets_item']    = 'item'
 default['osl-git']['secrets']['credentials'] = []
 
-default['osl-git']['ius_git_package'] = 'git224'
+default['osl-git']['ius_git_package'] = 'git226'
```
Comment (ramereth) [inline attributes/default.rb:5]: I might have typoed this, it needs to be `git236`

**Review by ramereth — CHANGES_REQUESTED** [review]

File: attributes/default.rb (hunk 6 above)
Comment (Justin-Fernbaugh) [reply attributes/default.rb]: Ah okay just pushed the changes to reflect the changes

File: test/integration/ius/inspec/ius_spec.rb [hunk 7]
```diff
@@ -2,7 +2,7 @@
   it { should_not be_installed }
 end
 
-describe package('git224') do
+describe package('git236') do
```
Comment (ramereth) [inline test/integration/ius/inspec/ius_spec.rb:5]: You need to fix the test below too please.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: test/integration/ius/inspec/ius_spec.rb (hunk 7 above)
Comment (Justin-Fernbaugh) [reply test/integration/ius/inspec/ius_spec.rb:5]: Updated the test for git version on git236

**Review by ramereth — APPROVED** [review]
Patch bump

**Comment by Justin-Fernbaugh** [comment]
!bump patch ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a patch-level version bump to v1.10.2.  Have a nice day!

**Pull request merged**